perf/benchmark/security/generate_policies/generate_policies.go

```diff
@@ -0,0 +1,185 @@
+		return "", err
+	}
+
+	authorizationPolicy, err := protomarshal.ToYAML(spec)
+	if err != nil {
+		return "", err
+	}
+
+	rulesYaml := bytes.Buffer{}
+	rulesYaml.WriteString("spec:\n")
+	scanner := bufio.NewScanner(strings.NewReader(authorizationPolicy))
+	for scanner.Scan() {
+		rulesYaml.WriteString(" " + scanner.Text() + "\n")
+	}
+	return string(headerYaml) + rulesYaml.String(), nil
+}
+
+func getOrderedKeySlice(ruleToOccurrences map[string]*ruleOption) *[]string {
+	var sortedKeys []string
+	for key := range ruleToOccurrences {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Strings(sortedKeys)
+	return &sortedKeys
+}
+
+func generateAuthorizationPolicy(action string, ruleToOccurrences map[string]*ruleOption, policy *MyPolicy) (string, error) {
+	spec := &authzpb.AuthorizationPolicy{}
+	switch action {
+	case "ALLOW":
+		spec.Action = authzpb.AuthorizationPolicy_ALLOW
+	case "DENY":
+		spec.Action = authzpb.AuthorizationPolicy_DENY
+	}
+
+	var ruleList []*authzpb.Rule
+	sortedKeys := getOrderedKeySlice(ruleToOccurrences)
+	for _, name := range *sortedKeys {
+		ruleOp := ruleToOccurrences[name]
+		rule := ruleOp.gen.generate(name, ruleOp.occurrence, action)
+		ruleList = append(ruleList, rule)
+	}
+	spec.Rules = ruleList
+
+	yaml, err := ToYAML(policy, spec)
+	if err != nil {
+		return "", err
+	}
+	return yaml, nil
+}
+
+func generateRule(action string, ruleToOccurrences map[string]*ruleOption,
+	policy *MyPolicy) (string, error) {
+
+	switch policy.Kind {
+	case "AuthorizationPolicy":
+		return generateAuthorizationPolicy(action, ruleToOccurrences, policy)
+	case "PeerAuthentication":
+		return "", fmt.Errorf("unimplemented")
+	case "RequestAuthentication":
+		return "", fmt.Errorf("unimplemented")
+	default:
+		return "", fmt.Errorf("invalid policy")
+	}
+}
+
+func createRules(action string, ruleToOccurrences map[string]*ruleOption, policy *MyPolicy) (string, error) {
+	yaml, err := generateRule(action, ruleToOccurrences, policy)
+	if err != nil {
+		return "", err
+	}
+	return yaml, nil
+}
+
+func createPolicyHeader(namespace string, name string, kind string) *MyPolicy {
+	metadata := MetadataStruct{namespace, name}
+	return &MyPolicy{
+		APIVersion: "security.istio.io/v1beta1",
+		Kind:       kind,
+		Metadata:   metadata,
+	}
+}
+
+func createRuleOptionMap(ruleToOccurancesPtr map[string]*int) *map[string]*ruleOption {
+	ruleOptionMap := make(map[string]*ruleOption)
+	for rule, occurrence := range ruleToOccurancesPtr {
+		ruleOptionMap[rule] = &ruleOption{}
+		ruleOptionMap[rule].occurrence = *occurrence
+		switch rule {
+		case "when":
+			ruleOptionMap[rule].gen = conditionGenerator{}
+		case "to":
+			ruleOptionMap[rule].gen = operationGenerator{}
+		case "from":
+			ruleOptionMap[rule].gen = sourceGenerator{}
+		default:
+			fmt.Println("invalid rules")
+		}
+	}
+	return &ruleOptionMap
+}
+
+func main() {
+	namespacePtr := flag.String("namespace", "twopods-istio", "Namespace in which the rule shall be applied to")
```

Comment:
do not hard code namespace? get it from environment variable

Reply:
same for other places.

Reply:
Makes sense. do you mean something like 
`os.Setenv("NAMESPACE", "twopods-istio")`
then using `os.Getenv("NAMESPACE")`

Reply:
This isn't hardcoding the namespace, its a default? So I think how it is is fine

Reply:
As John mentioned those are just the default values, by using a flag you can override them for example `-namespace=foo`

Reply:
@howardjohn  No, this is not a default, we can specify the Namespace ourselves. But we typically use "twopods-istio" that's why you think this is by default. I'd prefer to get it from env var: like what we did here: https://github.com/istio/tools/blob/master/perf/benchmark/runner/runner.py#L34

Reply:
I think John meant it's the default value if the flag `-namespace` is not specified in the command line, and yes you can override it to any value with `-namespace=foo` like Michael mentioned, so we can always override it when calling it from the runner.py.

Reply:
oh, yes, that's true.